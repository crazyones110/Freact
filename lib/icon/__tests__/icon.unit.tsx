import * as renderer from 'react-test-renderer'
import React from 'react'
import Icon from '../icon'
import {mount} from 'enzyme'

describe('icon组件', () => {
  it('render successfully', () => {
    const json = renderer.create(<Icon icon="bytedance"/>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('onClick', () => {
    const fn = jest.fn()
    const component= mount(<Icon icon="bytedance" onClick={fn}/>)
    component.find('svg').simulate('click')
    expect(fn).toBeCalled()
  })
})