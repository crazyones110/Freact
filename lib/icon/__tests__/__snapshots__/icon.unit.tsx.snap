// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon组件 render successfully 1`] = `
<svg
  className="freact-icon"
>
  <use
    xlinkHref="#bytedance"
  />
</svg>
`;
